libvast/include/vast/concept/parseable/vast/data.hpp

```diff
@@ -65,9 +72,7 @@ struct data_parser : parser_base<data_parser> {
       | parsers::duration
       | parsers::net
       | parsers::addr
-      | parsers::real
-      | parsers::count
-      | parsers::integer
+      | number_parser
```

Comment:
Maybe we can have `parser::number` here that returns a `data`? This would fit in nicer.